pkg/operator/ceph/provisioner/provisioner.go

```diff
@@ -119,9 +119,10 @@ func (p *RookVolumeProvisioner) Provision(options controller.VolumeOptions) (*v1
 					Driver: flexdriver,
 					FSType: p.provConfig.fstype,
 					Options: map[string]string{
-						flexvolume.StorageClassKey: storageClass,
-						flexvolume.PoolKey:         p.provConfig.pool,
-						flexvolume.ImageKey:        imageName,
+						flexvolume.StorageClassKey:     storageClass,
+						flexvolume.PoolKey:             p.provConfig.pool,
```

Comment:
i'm thinking we need to remove this concept of the `p.provConfig` member field completely.  i don't like it because it intends to capture per volume config, but it's scope is actually at the entire provisioner level.  This can lead to race conditions where if 2 volumes are being created at the same time, one can overwrite the others config accidentally, resulting in unintended consequences.

if we remove `provConfig` completely, we should be OK because with your change to also store namespace on the flexvolume options, all info we need persisted is done so on the flexvolume source.  There shoudln't be a need to store it on any member field of the provisioner anymore.

Does that sound right?

Reply:
Yeah, i was wondering that too. Just didn't want to assume that it is not needed given my novice status with rook. let me take a swing at completely deleting provConfig and see what happens.